atintegrators/BeamLoadingCavityPass.c

```diff
@@ -72,16 +74,23 @@ void BeamLoadingCavityPass(double *r_in,int num_particles,int nbunch,
     int *pslice;
     double *kz;
     double freqres = rffreq/(1-tan(vgenk[1])/(2*qfactor));
-    double vgen = vgenk[0];
-    double psi = vgenk[1];
-    
+    double vgen = 0.0;
+    double psi = 0.0;
+        
     for(i=0;i<nbunch;i++){
         tot_current += bunch_currents[i];
     }
 
+    if(cavitymode==2){
+        freqres = rffreq;
+    }else if(cavitymode==1){
+        vgen = vgenk[0];
+        psi = vgenk[1];
+    }
```

Comment:
Here you may raise an error if the mode is unknown, or even in the initialization below where there are already sanity check

Reply:
done